### PR TITLE
Add distances to database

### PR DESCRIPTION
**Pull request by Roger-Chavez**
Closes #(issue)

### Key features
* CSV data for distances are being treated as undirected edges so removed duplicate edges
* CSVs have been updated to reflect the stadium id changes of MLBInfo in master
* Database now contains the distances as a vector of tuple (toStadiumId, fromStadiumId, distance)
* Database now contains a map of coordinates
* getters have been made to return a const ref of these containers
### Future Improvements
* N / A 

### Notes
* Load and save have been updated to reflect changes

### CODE REVIEW

File: src/datastore/DistanceBetweenExpansionStadium.csv [hunk 1]
```diff
@@ -1,4 +1,4 @@
 # id, stadium name, id, stadium name, distance
-80,Las Vegas Stadium,52,Oakland-Alameda County Coliseum,325
-80,Las Vegas Stadium,55,Dodger Stadium,300
-80,Las Vegas Stadium,61,Chase Field,250
\ No newline at end of file
+80,Las Vegas Stadium,69,Oakland-Alameda County Coliseum,325
+80,Las Vegas Stadium,63,Dodger Stadium,300
+80,Las Vegas Stadium,50,Chase Field,250
```
Comment (ChrisRzech) [inline src/datastore/DistanceBetweenExpansionStadium.csv:4]: Why were the stadium IDs changed?

File: src/datastore/DistanceBetweenStadiums.csv [hunk 2]
```diff
@@ -1,107 +1,54 @@
...
-75,Tropicana Field,73,Marlins Park,210
-75,Tropicana Field,57,Minute Maid Park,790
-56,Wrigley Field,70,Guaranteed Rate Field,50
-56,Wrigley Field,79,Miller Park,80
-56,Wrigley Field,60,Kauffman Stadium,415
-66,Yankee Stadium,78,PNC Park,315
-66,Yankee Stadium,67,Citizens Bank Park,80
-66,Yankee Stadium,64,Citi Field,50
\ No newline at end of file
+# id, stadium name, id, stadium name, distance
```
Comment (ChrisRzech) [inline src/datastore/DistanceBetweenStadiums.csv:1]: What happened in this file?

File: src/datastore/database.cpp [hunk 3]
```diff
@@ -215,3 +168,102 @@ void Database::saveToFile(const std::string& path)
...
+                                                    std::stoi(distance[4])));
+    }
+}
+
+/**
+ * This method returns all the teams
+ *
+ * @return teams is returned
+ */
+nstd::map<int,Team> Database::getTeams()
```
Comment (ChrisRzech) [inline src/datastore/database.cpp:190]: Why were all these functions moved from their original position?

File: src/datastore/database.hpp [hunk 4]
```diff
@@ -18,16 +18,23 @@
 class Database
 {
 public:
+    /* Aliases */
+    using completedEdge = std::vector<std::tuple<int,int,int>>;  //"from" <-> "to" and distance
```
Comment (ChrisRzech) [inline src/datastore/database.hpp]: 1 complete edge = a vector? doesn't make sense

**Review by ChrisRzech — CHANGES_REQUESTED** [review]
Where is the graph data-structure?
Why so many CSV file changes? There are more changes to them than necessary, and you didn't provide a description in the PR description about why you made them.

File: src/datastore/database.hpp (hunk 4 above)
Comment (Roger-Chavez) [reply src/datastore/database.hpp]: I'll make the tuple a completed edge

File: src/datastore/database.cpp (hunk 3 above)
Comment (Roger-Chavez) [reply src/datastore/database.cpp:190]: Moved the methods to be in the same order as the header file

File: src/datastore/DistanceBetweenStadiums.csv (hunk 2 above)
Comment (Roger-Chavez) [reply src/datastore/DistanceBetweenStadiums.csv:1]: Changed id's to reflect changes in MLBInfo file

File: src/datastore/DistanceBetweenExpansionStadium.csv (hunk 1 above)
Comment (Roger-Chavez) [reply src/datastore/DistanceBetweenExpansionStadium.csv:4]: updated to match MLBInfo file

File: src/datastore/DistanceBetweenStadiums.csv (hunk 2 above)
Comment (ChrisRzech) [reply src/datastore/DistanceBetweenStadiums.csv:1]: What happened to half of the file? It's like Thanos came in here and snapped his fingers lol

**Comment by Roger-Chavez** [comment]
> Where is the graph data-structure?
> Why so many CSV file changes? There are more changes to them than necessary, and you didn't provide a description in the PR description about why you made them.

I thought we agreed I was adding the distance data to the database, mainly what I wanted to do was just add the coordinates, but figured I'd add the distance data while I was at it. Need the coordinates for the main pr I'm trying to land by Monday. We never agreed on adding a graph, all I was adding was the distances so that when the graph is added it can just grab the distances from the database. The only things that changed form the CSV were the coordinates, updating the stadium ids to increment with the team and we also agreed to treat the edges as undirected so removed duplicate edges. I originally put the ids on distances and the id's for the MLBInfo were out of order, I went back and redid it. None of the data that Lebo provided was altered.

File: src/datastore/DistanceBetweenStadiums.csv (hunk 2 above)
Comment (Roger-Chavez) [reply src/datastore/DistanceBetweenStadiums.csv:1]: I've discussed all of these changes with you already, I asked if you wanted me to remove duplicate edges that just went the other way

File: src/datastore/DistanceBetweenExpansionStadium.csv (hunk 1 above)
Comment (ChrisRzech) [reply src/datastore/DistanceBetweenExpansionStadium.csv:4]: Why were they changed in the first place?

File: src/datastore/DistanceBetweenStadiums.csv (hunk 2 above)
Comment (ChrisRzech) [reply src/datastore/DistanceBetweenStadiums.csv:1]: But you don't address the changes in the PR description.

File: src/datastore/database.cpp [hunk 5]
```diff
@@ -222,3 +160,102 @@ void Database::saveToFile(const std::string& path)
...
+    return stadiums;
+}
+
+/**
+ * Finds team by it's id.
+ *
+ * @param id an int id used to find a Team
+ * @return A const reference to a Team object
+ */
+Team& Database::findTeamById(int id)
```
Comment (ChrisRzech) [inline src/datastore/database.cpp:211]: Why are these functions changed from what was on master?

**Review by ChrisRzech — CHANGES_REQUESTED** [review]
Add more descriptions to the PR description.

File: src/datastore/database.cpp (hunk 5 above)
Comment (Roger-Chavez) [reply src/datastore/database.cpp:211]: These changes are in master

**Review by yosmo78 — APPROVED** [review]
When you merge with master, as long as those findById funcs don't get changed then I approve

**Review by ChrisRzech — APPROVED** [review]
👍 let's land this baby

**Pull request merged**